### DISCUSSION TITLE
Bazel Version Constraints in BCR

### DISCUSSION THREAD

**Discussion by shantanu-gontia** [bazelbuild/bazel-central-registry#2336] [Q&A] [answered]
Hi, I was looking into adding LLVM18 to the BCR. I noticed there's already an entry for llvm-project 17.0.3, so I tried getting it to work. But apparently, it doesn't work with Bazel Versions >= 7.0 (I think this is because `exec_tools` no longer works in genrules).

But there was no way to know this from the BCR entry. Is it possible/worthwhile to let us know which Bazel versions a module is supported on. Or, even to complain that the module might not work with your version of Bazel.

**Comment by keith** [comment, marked as answer]
Theoretically folks should be annotating their MODULE.bazel with this compatibility:

https://github.com/bazelbuild/bazel-central-registry/blob/e5914f44e066666c3c18cc6fd992c3e8e76fb2d9/modules/rules_apple/3.1.0/MODULE.bazel#L4

In which case you would be able to see it there. And it would fail earlier than some random bazel API change. I think thus far folks have been pretty lax on that, and it doesn't help you spot the case when a transitive dependency does encode this requirement and it hits you too. I would expect this to improve as more folks use bzlmod

**Reply by shantanu-gontia** [reply]
That's great. Exactly what I was looking for. Moving forward, can (should?) this be enforced for new Modules?

**Reply by fmeum** [reply]
We could require a `bazel_compatibility` argument if a module doesn't list Bazel 6 in its presubmit.yml. but we can't modify module files after release, so incompatibilities with future bazel versions are hard to track.

@meteorcloudy

**Reply by meteorcloudy** [reply]
I think that requirement makes sense. But "Bazel 6" should be "the last maintained major release that officially supports Bzlmod" (https://bazel.build/release#support-matrix) since we probably don't want to require testing against a deprecated Bazel version.
